File: server/app/models/db/user/user_course_progress_response.py
from pydantic import BaseModel
from typing import Dict
from app.db.enums import AvailableCourse, AvailableDialect, AvailableLanguage


class UserCourseProgressResponse(BaseModel):
    id: int
    course_name: AvailableCourse
    language: AvailableLanguage
    dialect: AvailableDialect | None = None
    default_dialect: AvailableDialect | None = None
    total_modules: int
    curr_module: int
    covered_words: Dict[str, int]
    problem_counter: int
    current_vocab_problem_set: int

File: client/types/response_models/UserCourseProgressResponse.ts
import { AvailableCourse, AvailableDialect, AvailableLanguage } from "../enums";

export interface UserCourseProgressResponse {
    id: number;
    course_name: AvailableCourse;
    language: AvailableLanguage;
    dialect: AvailableDialect | null;
    default_dialect: AvailableDialect | null;
    total_modules: number;
    curr_module: number;
    covered_words: Record<string, number>;
    problem_counter: number;
    current_vocab_problem_set: number;
}